packages/flutter_tools/test/android/gradle_test.dart

```diff
@@ -126,4 +128,92 @@ someOtherProperty: someOtherValue
       expect(project.assembleTaskFor(const BuildInfo(BuildMode.release, 'unknown')), isNull);
     });
   });
+
+  String _propertyFor(String key, File file) {
+    return file
+        .readAsLinesSync()
+        .where((String line) => line.startsWith('$key='))
+        .map((String line) => line.split('=')[1])
+        .first;
+  }
+
+  group('gradle local.properties', () {
+    Directory temp;
+
+    setUp(() {
+      Cache.disableLocking();
+      temp = fs.systemTempDirectory.createTempSync('flutter_tools');
+    });
+
+    tearDown(() {
+      temp.deleteSync(recursive: true);
+    });
+    testUsingContext('test version from pubspec.yaml', () async {
+      const BuildInfo buildInfo = const BuildInfo(BuildMode.release, null);
+      final String projectPath = await createProject(temp);
+
+      try {
+        await updateLocalProperties(projectPath: projectPath, buildInfo: buildInfo);
+
+        final String propertiesPath = fs.path.join(projectPath, 'android', 'local.properties');
+        final File localPropertiesFile = fs.file(propertiesPath);
+
+        expect(_propertyFor('flutter.versionName', localPropertiesFile), '1.0.0');
+        expect(_propertyFor('flutter.versionCode', localPropertiesFile), '1');
+      } on Exception {
+        // Android SDK not found, skip test
+      }
+    });
+
+    testUsingContext('test buildName from build command', () async {
```

Comment:
Duplication can be removed by defining a function like
```dart
checkLocalProperties({
  String projectPath,
  BuildInfo buildInfo,
  String expectedVersionName,
  String expectedVersionCode,
}) { ... }
```

Reply:
done